### PR TITLE
test: speed up test workflow

### PR DESCRIPTION
**Pull request by afc163**
[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE_CN.md?plain=1)

### 🤔 This is a ...

- [ ] 🆕 New feature
- [ ] 🐞 Bug fix
- [ ] 📝 Site / documentation improvement
- [ ] 📽️ Demo improvement
- [ ] 💄 Component style improvement
- [ ] 🤖 TypeScript definition improvement
- [ ] 📦 Bundle size optimization
- [ ] ⚡️ Performance optimization
- [ ] ⭐️ Feature enhancement
- [ ] 🌐 Internationalization
- [ ] 🛠 Refactoring
- [ ] 🎨 Code style optimization
- [x] ✅ Test Case
- [ ] 🔀 Branch merge
- [ ] ⏩ Workflow
- [ ] ❓ Other (about what?)

### 🔗 Related Issues

> - Describe the source of related requirements, such as links to relevant issue discussions.
> - For example: close #xxxx, fix #xxxx

### 💡 Background and Solution

> - The specific problem to be addressed.
> - List the final API implementation and usage if needed.
> - If there are UI/interaction changes, consider providing screenshots or GIFs.

https://github.com/ant-design/ant-design/actions/runs/10336878232?pr=50346

- [x] 优化了 workflow 的逻辑依赖关系，提前了代码覆盖率的 job。
- [x] 移除了一些没必要的检测项，比如 check_metadata
- [x] 升级到 node 20
- [x] 补齐了一些检测项，比如 es 目录的检测。

| before | after |
| --- | --- |
| 图片 | 图片 |

### 📝 Change Log

> - Describe the impact of the changes on developers, not the solution approach.
> - Reference: https://ant.design/changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |      --     |
| 🇨🇳 Chinese |       --    |

### CODE REVIEW

**Review by li-jia-nan — DISMISSED** [review]

File: tests/dekko/es.test.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+const $ = require('dekko');
+const chalk = require('chalk');
+
+$('es').isDirectory().hasFile('index.js').hasFile('index.d.ts');
+
+$('es/*')
+  .filter(
+    (filename) =>
```
Comment (li-jia-nan) [inline tests/dekko/es.test.js:8]:
```suggestion
    ['index.js', 'index.d.ts', '.map'].every(suffix => !filename.includes(suffix))
```

File: tests/dekko/es.test.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+      !filename.endsWith('index.js') &&
+      !filename.endsWith('index.d.ts') &&
+      !filename.endsWith('.map'),
+  )
+  .isDirectory()
+  .filter(
+    (filename) =>
```
Comment (li-jia-nan) [inline tests/dekko/es.test.js:15]:
```suggestion
    ['style', '_util', 'locale'].every(suffix => !filename.includes(suffix))
```

File: tests/dekko/es.test.js (hunk 1 above)
Comment (afc163) [reply tests/dekko/es.test.js:8]: 不能这么改吧。

File: tests/dekko/es.test.js (hunk 2 above)
Comment (afc163) [reply tests/dekko/es.test.js:15]: 同上。

File: tests/dekko/es.test.js (hunk 1 above)
Comment (afc163) [reply tests/dekko/es.test.js]: 改好了。

**Review by li-jia-nan — APPROVED** [review]

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -198,6 +198,7 @@ jobs:
     name: test-react-latest-dist
     strategy:
       matrix:
+        module: ['dist', 'dist-min']
```
Comment (afc163) [inline .github/workflows/test.yml:201]: dist 有点用，不能删，不然看不到报错的代码行。

**Pull request merged**